src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java

```diff
@@ -138,7 +140,7 @@ public BsonValue toBsonValue(final DataType dataType, final Object data) {
           || DataTypes.TimestampType.acceptsType(dataType)) {
         return new BsonDateTime(((Date) data).getTime());
       } else if (DataTypes.NullType.acceptsType(dataType)) {
```

Comment:
In the case of data type `Array<String>` eg: `["a", null]` the string value `a` would have been fine but it would have tried to convert a `null` value to string eg: `new BsonString((String) data)`.  That causes the BsonString constructor to throw an exception that is then caught and wrapped in a DataException below.